.github/workflows/pr.ci.js.yml

```
@@ -0,0 +1,37 @@
+# This workflow will do a clean install of node dependencies and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: exercism/javascript pull-request/ci
+
+on: [push, pull_request]
```

```suggestion
on: pull_request
```

I think if you set this to only `pull_request`, you can drop the `if`s below

---

Is there a way where we can still run push, but only in PRs to be merged in?

Push on PRs runs on the post merge code. I think.

---

I'm not sure what you mean. Do you want it to run after the merge button was clicked but before the commit is merged into the main branch?

---

No, when you do `[push, pull_request]`, you'll normally see your action twice on GitHub.

1. the action will run on the unmerged code (this is what works with the current code)
2. the action will run on a non-existing commit that merges the PR into the target branch.

I'm wondering if we can still achieve number 2, with only the changed files.

---

(I've changed it to PR check only for now, since CI check runs after merge, which is fine for now)